packages/batch-submitter/src/batch-submitter/batch-submitter.ts

```diff
@@ -94,7 +98,7 @@ export abstract class BatchSubmitter {
     return this._submitBatch(range.start, range.end)
   }
 
-  protected async _checkBalance(): Promise<void> {
+  protected async _hasEnoughETHToCoverGasCosts(): Promise<boolean> {
```

Comment:
Nit, I think this could be refactored to not be a class function. Having it be a class function makes it hard to test.